### PR TITLE
fix/alert/alert-list-ui-tests : Fix and complete UI tests for AlertListScreen

### PR DESCRIPTION
**Pull request by charliemangano**
# Fix and complete UI tests for AlertListScreen
## Description
This PR introduces corrects some duplicated testing for the AlertList screen. It closes issue #69 and fixes part of #63.
## Changes
Added dependencies to make mockito work in androidTests, and properly mocked `NavigationActions`. Refactored some of the tests to not duplicate testing purposelessly.
#### Added
None
#### Modified
- `app/build.gradle.kts` and `gradle/libs.versions.toml`: add the missing dependency
- `app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt`: modified some tests
#### Removed
None
## Dependencies Added
- [`com.linkedin.dexmaker:dexmaker-mockito-inline`](https://stackoverflow.com/a/67337292) to make mockito work with androidTests. Had to remove `libs.mockito.android` for androidTests as a consequence
## Testing
All UI tests for `AlertListScreen` pass. They should be complete enough.

### CODE REVIEW

File: app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt [hunk 1]
```diff
@@ -7,23 +7,33 @@ import androidx.compose.ui.test.assertTextEquals
 import androidx.compose.ui.test.junit4.createComposeRule
```
Comment (taghizadlaura) [inline app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt:7]: All tests LGTM. I don't have any suggestions to improve it.

**Review by taghizadlaura — APPROVED** [review]
## Summary
LGTM
The `AlertListScreenTest` class is a well-structured unit test suite for the `AlertListScreen` component of the PeriodPals app, effectively utilizing Jetpack Compose testing frameworks and Mockito to ensure the correct functioning of UI elements and interactions.

## Important
#### Code Quality
- **Readability**: The code is organized and method names are descriptive, making it easy to understand the purpose of each test.
- **Consistency**: The use of annotations and assertions follows best practices, contributing to a uniform structure throughout the test suite.

#### Functionality
- **Correctness**: Tests cover a wide range of scenarios, verifying that UI elements are displayed correctly and respond as expected.
- **Edge Cases**: The suite addresses interactions, ensuring that actions such as tab switching and clicking on alerts do not lead to unintended behavior.

#### Performance
- Not applicable in this context.

#### Security
- Not applicable in this context.

## Testing
The tests effectively cover the functionality of the `AlertListScreen`, including the initial state, tab interactions, and the display of content. The use of assertions is robust, confirming that the UI behaves as expected under various conditions.

## Documentation
The class is well-documented through clear test names, which indicate the purpose of each test. 

## Suggestions
Nothing to add

## Steps before PR approved
LGTM

**Review by lazarinibruno — APPROVED** [review]

**Review by lazarinibruno — APPROVED** [review]
## Summary

Thanks for improving the tests for this screen!

- The code is cleaner now
- Navigation is finally properly mocked
- You wisely removed redundant checks that I wrote while trying to get the branch coverage above 80% (ㅠ﹏ㅠ)
- Tests run and pass locally 🔥
  

## Steps before merge

LGTM!

File: app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt [hunk 2]
```diff
@@ -7,23 +7,33 @@ import androidx.compose.ui.test.assertTextEquals
...
 @RunWith(AndroidJUnit4::class)
 class AlertListScreenTest {
 
+  private lateinit var navigationActions: NavigationActions
   @get:Rule val composeTestRule = createComposeRule()
 
+  @Before
+  fun setUp() {
+    navigationActions = mock(NavigationActions::class.java)
+    `when`(navigationActions.currentRoute()).thenReturn(Route.ALERT)
```
Comment (lazarinibruno) [inline app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt:31]: should be: 

```
 `when`(navigationActions.currentRoute()).thenReturn(Route.ALERT_LIST)
```
